cmd/zgenhostid/zgenhostid

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Emulate genhostid(1) available on RHEL/CENTOS, for use on SUSE linux systems
+# which do not have that utility.
+#
+# Usage:
+#    genhostid
+#    genhostid <value>
+#
+# If /etc/hostid already exists, the script exits immediately and changes
+# nothing.  Unlike genhostid, this generates an error message.
+#
+# The first form generates a random hostid and stores it in /etc/hostid.
+# The second form checks that the provided value is between 0x1 and 0xFFFFFFFF
+# and if so, stores it in /etc/hostid.  This form is not supported by
+# genhostid(1).
+
+hostid_file=/etc/hostid
+
+function usage {
+	echo "$0 [value]"
+	echo "If $hostid_file is not present, store a hostid in it." >&2
+	echo "The optional value must be an 8-digit hex number between" >&2
+	echo "1 and 2^32-1.  If no value is provided, a random one will" >&2
+	echo "be generated.  The value must be unique among your systems." >&2
+}
+
+if [ -e $hostid_file ]; then
+	echo "$hostid_file already exists.  No change made." >&2
+	exit 1
+fi
+
+if [ -n "$1" ]; then
+	host_id=$1
+else
+	# $RANDOM goes from 0..32k-1
+	number1=$(( ( (RANDOM%4)*32768*32768 ) ))
+	number2=$(( ( RANDOM*32768) ))
+	number3=$(( ( RANDOM) ))
```

Comment:
That's a good point that the libc-generated one may already be in use.  

My concern is that someone will either have a poorly chosen name-to-address mapping, so that two machines end up with the same libc-generated hostid; or they have something like corosync in place that changes the IP address as part of a failover mechanism.

Do you think those are not meaningful risks?

And, it's already possible to get the same effect with:
`
zgenhostid $(hostid)
`
Although that should probably be in the manpage.